ambari-server/src/main/resources/common-services/ATLAS/0.1.0.2.3/package/scripts/metadata.py

```diff
@@ -149,9 +151,26 @@ def metadata(type='server'):
                                        roles = [params.infra_solr_role_atlas, params.infra_solr_role_ranger_audit, params.infra_solr_role_dev],
                                        new_service_principals = [params.atlas_jaas_principal])
 
-      create_collection('vertex_index', 'atlas_configs', jaasFile)
-      create_collection('edge_index', 'atlas_configs', jaasFile)
-      create_collection('fulltext_index', 'atlas_configs', jaasFile)
+      if default('configurations/infra-solr-env/infra_solr_ssl_enabled', False):
```

Comment:
i think ranger can run this command against external Solr as well, so maybe you can try out to just pass these truststore fields anyway (if they are not empty) and test that it cause any problem if solr is not using ssl (i guess it wont cause problem but lets check it)

Reply:
private static final String TRUSTSTORE_LOCATION_ARG = "javax.net.ssl.trustStore";
  private static final String TRUSTSTORE_PASSWORD_ARG = "javax.net.ssl.trustStorePassword";
  private static final String TRUSTSTORE_TYPE_ARG = "javax.net.ssl.trustStoreType";

These properties map to the java ssl connection client. If they are passed and there is no SSL connection happening it wont hurt anyways. 

I guess we can adjust it so it does it for both external and internal, or have it done by default if the property exists

Reply:
Yes this will be needed to handle externally managed solr for Ranger Service

Reply:
I have changed it to supply the truststore information if they are defined regardless of the status of solr SSL/nonSSL Internal/External